engine/src/main/java/org/terasology/rendering/world/WorldRendererImpl.java

```diff
@@ -634,7 +634,8 @@ private void preRenderUpdate(RenderingStage renderingStage) {
         renderableWorld.queueVisibleChunks(isFirstRenderingStageForCurrentFrame);
 
         if (requestedTaskListRefresh) {
-            renderTaskListGenerator.refresh();
+            List<Node> orderedNodes = renderGraph.getNodesInTopologicalOrder();
+            renderPipelineTaskList = renderTaskListGenerator.generateFrom(orderedNodes);
```

Comment:
> They feel like they are the competence of the generator rather than the renderer.
I'm not sure I follow you here.

> But I guess there's a good reason to have them here and there?
As it turns out, there wasn't :D

Reply:
For what I can see those two lines should continue to be wrapped by the generator's refresh() method and a call to refresh() should be made also in initRenderGraph().